pkg/manager/manager.go

```diff
@@ -314,6 +314,19 @@ type LeaderElectionRunnable interface {
 	NeedLeaderElection() bool
 }
 
+// WarmupRunnable knows if a Runnable should be a warmup runnable. A warmup runnable is a runnable
+// that should be run when the manager is started, but before the leader election is acquired.
+// Note: Implementing this interface is only useful when LeaderElection is enabled, as the behavior
+// without leaderelection is to run LeaderElectionRunnables immediately.
+type WarmupRunnable interface {
```

Comment:
Do we want to export this interface?

If we only need it for controller we could probably keep it unexported for now (like "hasCache")

Also allows us refactorings going forward if necessary

Reply:
Addressed in https://github.com/kubernetes-sigs/controller-runtime/pull/3192/commits/6bb4616ba3e651c98015a5382549982074f563ac